code/modules/reagents/reagent_containers/mortar.dm

```diff
@@ -0,0 +1,62 @@
+/obj/item/chems/glass/mortar
+	name = "mortar"
+	desc = "A hard, sturdy bowl used to hold organic matter for crushing."
+	icon = 'icons/obj/items/chem/mortar.dmi'
+	volume = 40
+	material = /decl/material/solid/stone/basalt
+	material_alteration = MAT_FLAG_ALTERATION_ALL
+
+// todo: generalize to use TOOL_PESTLE
+/obj/item/chems/glass/mortar/attack_self(mob/user)
+	var/list/contained_atoms = get_contained_external_atoms()
+	var/obj/item/crushing_item = LAZYACCESS(contained_atoms, 1)
+	if(crushing_item)
+		crushing_item.dropInto(loc || get_turf(user))
+		return TRUE
+	return ..()
+
+/obj/item/chems/glass/mortar/attackby(obj/item/using_item, mob/living/user)
+	if((. = ..()))
+		return
+	if(!istype(using_item))
+		return FALSE
+	if(!CanPhysicallyInteract(user) || !user.check_dexterity(DEXTERITY_HOLD_ITEM))
+		return TRUE
+	var/list/contained_atoms = get_contained_external_atoms()
+	var/obj/item/crushing_item = LAZYACCESS(contained_atoms, 1)
+	var/decl/material/attacking_material = using_item.get_material()
+	var/decl/material/crushing_material = crushing_item?.get_material()
+	var/skill_factor = CLAMP01(1 + 0.3*(user.get_skill_value(SKILL_CHEMISTRY) - SKILL_EXPERT)/(SKILL_EXPERT - SKILL_MIN))
+	if(!crushing_item)
+		if(!using_item.reagents?.total_volume)
+			to_chat(user, SPAN_NOTICE("\The [using_item] is not suitable for grinding."))
+			return TRUE
+		if(!user.try_unequip(using_item, src))
+			return TRUE
+		to_chat(user, SPAN_NOTICE("You add \the [using_item] to \the [src] for grinding."))
+		return TRUE
+	if(using_item.force <= 0 || !attacking_material || !crushing_material)
```

Comment:
Do plants or pills have a material set? Wouldn't be able to grind them without that it seems

Reply:
`material = /decl/material/solid/organic/plantmatter` for pills, for some reason